proto/protovalidate/buf/validate/BUILD.bazel

```diff
@@ -42,3 +42,16 @@ cc_proto_library(
     visibility = ["//visibility:public"],
     deps = [":validate_proto"],
```

Comment:
You will need to add the `expression_proto` here

Reply:
I am not so sure 🤔 `:expression_proto` is already in the dependencies of `:validate_proto` so it should work correctly as is, no?

Reply:
That won't generate for dependencies we will need a second `cc_proto_library` rule for generating the corresponding cc files for `expression_proto`. That should be a dependency for this library.

Output of `bazel build //proto/protovalidate/buf/validate:validate_proto_cc`:

```sh
Target //proto/protovalidate/buf/validate:validate_proto_cc up-to-date:
  bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.h
  bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.cc
  bazel-bin/proto/protovalidate/buf/validate/libvalidate_proto.a
```
Notice how it only generates validate files.

Reply:
`cc_proto_library` supports only a singular `proto_library` dependency, so I cannot add `:expression_proto` directly there.

But, I can add a `cc_proto_library` and add it to rules that depend on `validate_proto_cc`.

Reply:
Sorry I meant the new `cc_proto_library` should be the dependency.

Reply:
But `cc_proto_library` supports only a single `proto_library` dependency, with no other dependencies. If I try to pass an extra `cc_proto_library` as a dependency, the build will fail.

Honestly, I feel like the extra `cc_proto_library` is not helping.

With the previous build file (without the `expression_proto_cc` rule), while it is true that `bazel build` lists only the direct targets, I do see the dependency proto `c` code being generated:

```
$ bazel build //proto/protovalidate/buf/validate:validate_proto_cc
INFO: Streaming build results to: https://bornite.cluster.engflow.com/invocation/577313d5-97e4-4a5e-b8a4-1dd816f3940a
INFO: Analyzed target //proto/protovalidate/buf/validate:validate_proto_cc (1 packages loaded, 5 targets configured).
INFO: Found 1 target...
Target //proto/protovalidate/buf/validate:validate_proto_cc up-to-date:
  bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.h
  bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.cc
  bazel-bin/proto/protovalidate/buf/validate/libvalidate_proto.a
  bazel-bin/proto/protovalidate/buf/validate/libvalidate_proto.so
INFO: Elapsed time: 5.506s, Critical Path: 4.76s
INFO: 27 processes: 19 internal, 8 linux-sandbox.
INFO: Streaming build results to: https://bornite.cluster.engflow.com/invocation/577313d5-97e4-4a5e-b8a4-1dd816f3940a
INFO: Build completed successfully, 27 total actions

$ ls bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/
expression_proto  validate_proto

$ ls bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/expression_proto/buf/validate/expression.p*
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/expression_proto/buf/validate/expression.pb.cc
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/expression_proto/buf/validate/expression.pb.h
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/expression_proto/buf/validate/expression.proto

$ ls bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.p*
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.cc
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.h
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.proto
```

Reply:
But `cc_proto_library` supports only a single `proto_library` dependency, with no other dependencies. If I try to pass an extra `cc_proto_library` as a dependency, the build will fail.

Honestly, I feel like the extra `cc_proto_library` is not helping.

With the previous build file (without the `expression_proto_cc` rule), while it is true that `bazel build` lists only the direct targets, I do see the dependency proto `c` code being generated:

```
$ bazel build //proto/protovalidate/buf/validate:validate_proto_cc
INFO: Streaming build results to: ###
INFO: Analyzed target //proto/protovalidate/buf/validate:validate_proto_cc (1 packages loaded, 5 targets configured).
INFO: Found 1 target...
Target //proto/protovalidate/buf/validate:validate_proto_cc up-to-date:
  bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.h
  bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.cc
  bazel-bin/proto/protovalidate/buf/validate/libvalidate_proto.a
  bazel-bin/proto/protovalidate/buf/validate/libvalidate_proto.so
INFO: Elapsed time: 5.506s, Critical Path: 4.76s
INFO: 27 processes: 19 internal, 8 linux-sandbox.
INFO: Streaming build results to: ###
INFO: Build completed successfully, 27 total actions

$ ls bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/
expression_proto  validate_proto

$ ls bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/expression_proto/buf/validate/expression.p*
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/expression_proto/buf/validate/expression.pb.cc
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/expression_proto/buf/validate/expression.pb.h
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/expression_proto/buf/validate/expression.proto

$ ls bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.p*
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.cc
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.pb.h
bazel-bin/proto/protovalidate/buf/validate/_virtual_imports/validate_proto/buf/validate/validate.proto
```